src/Shared/OsmMapData.m

```diff
@@ -2322,33 +2343,39 @@ -(BOOL)discardStaleData
 // after uploading a changeset we have to update the SQL database to reflect the changes the server replied with
 -(void)updateSql:(NSDictionary *)sqlUpdate
 {
-	NSMutableArray * insertNode		= [NSMutableArray new];
-	NSMutableArray * insertWay		= [NSMutableArray new];
-	NSMutableArray * insertRelation	= [NSMutableArray new];
-	NSMutableArray * deleteNode		= [NSMutableArray new];
-	NSMutableArray * deleteWay		= [NSMutableArray new];
-	NSMutableArray * deleteRelation	= [NSMutableArray new];
+	NSMutableArray<OsmNode *> * insertNode		= [NSMutableArray new];
+	NSMutableArray<OsmWay *> * insertWay		= [NSMutableArray new];
+    NSMutableArray<OsmRelation *> * insertRelation    = [NSMutableArray new];
+	NSMutableArray<OsmNode *> * deleteNode		= [NSMutableArray new];
+	NSMutableArray<OsmWay *> * deleteWay		= [NSMutableArray new];
+	NSMutableArray<OsmRelation *> * deleteRelation	= [NSMutableArray new];
 	[sqlUpdate enumerateKeysAndObjectsUsingBlock:^(OsmBaseObject * object, NSNumber * insert, BOOL *stop) {
 		if ( object.isNode ) {
 			if ( insert.boolValue )
-				[insertNode addObject:object];
+				[insertNode addObject:object.isNode];
```

Comment:
The reason why we need to change these lines here is that the `NSMutableArray` instances only accept the hinted types. If we did not change these lines, Xcode displays a warning.

I do realize that this is not very intuitive, but it's the quickest approach to this. Let me know what you think!